src/flow_graph.py

```diff
@@ -1096,15 +1274,163 @@ def reset_block_info(self) -> None:
             node.block.block_info = None
 
 
+def phi_reg_sources(node: Node, reg: Access, imdom_srcs: RefSet) -> RefSet:
```

Comment:
So, we're presuming that `reg` references a static location here, e.g. a register, a stack slot, or a global variable. (But we shouldn't track global variables as phis and imo we shouldn't do alias analysis at this level so let's forget about the last point.) But Access is more general than that -- it supports "reg + whatever" -- and using that here would be incorrect. I feel like we should drop that generalized representation and change to a Union[Register, StackRef(int), MemoryAccess]. (sp + non constant is a MemoryAccess, similar to how computing sp + non constant with an add instruction and then referencing the resulting memory would work. We can relatively safely assume no aliasing I think)

Reply:
(Also, re my comments about -framepointer from before that turn fp into an alias for sp, we should just do that analysis in an earlier pass than translate to the extent that we care about supporting it)